### DISCUSSION TITLE
How do I access the execution time in a transient prompt?

### DISCUSSION THREAD

**Discussion by miversen33** [JanDeDobbeleer/oh-my-posh#5432] [Troubleshoot] [answered]
Looking at the documentation, its a bit vague on this. It appears like I have 2 options
1) [Cross Segment Template Properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties), where I store the execution time as a segment in prompt and then pull it for the transient prompt. The issue with this is it appears to be off by one, showing the previous commands execution in the current transient prompt. Example
![image](https://github.com/user-attachments/assets/f7e5e617-dd80-4404-b030-1991d0b31f3d)  
2) This one is much more vague. It was mentioned by JanDeDobbeleer in #2605 that they were going to setup a cache for accessing the command execution information. Indeed, [this cache does exist](https://ohmyposh.dev/docs/configuration/templates#global-properties), however it doesn't have the execution time listed as an available option (though this was hinted at being a reason for caching)
>  the transient prompt's template can already get a lot of information available from the standard [templates](https://ohmyposh.dev/docs/configuration/templates#global-properties), moving that to a full prompt is a bit beyond the scope of the transient prompt. This means you have access to the exit code, date/time (through [sprig](http://masterminds.github.io/sprig/date.html)). **The only missing piece is execution time which needs to be cached anyways as it's not available/correct when the transient prompt is invoked. That makes things fast, as you don't want to wait on 2 prompts being rendered.**

Is there a way to do what I am trying or am I SOL here?

Thanks in advance!

**Comment by JanDeDobbeleer** [comment, marked as answer]
@miversen33 you will never see the execution time on the same line as the executed command as transient is invoked before the command is executed. Your screenshot is correct. You will always see the execution time result in the prompt below the command. This has nothing to do with caching, it's how it works.

**Reply by miversen33** [reply]
That makes perfect sense. Thanks!

**Reply by FerrarioChristian** [reply]
> @miversen33 you will never see the execution time on the same line as the executed command as transient is invoked before the command is executed. Your screenshot is correct. You will always see the execution time result in the prompt below the command. This has nothing to do with caching, it's how it works.

I guess the same thing apply for the exit .Code color then? I can't seem to find a way to make transient prompt color based on the command success/failure so I'm looking for a confirm. If that's not the right place to ask, sorry in advance, please let me know.
Edit: I found this: https://ohmyposh.dev/blog/whats-new-2#transient-prompt, but I can't make it work as expected.

**Reply by JanDeDobbeleer** [reply]
@FerrarioChristian `.Code` is available in transient, you can use it to define colors using color templates as well.

**Comment by dperson** [comment]
Maybe I'm just being dumb, but I'm not getting any data from `.Segments.ExecutionTime.Ms` or `.Segments.ExecutionTime.FormattedMs`. How do I use the [Cross Segment Template Properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties) to access them?

I've read the [Execution Time](https://ohmyposh.dev/docs/segments/system/executiontime) segment page, and the information on the [Cross Segment Template Properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties), but if those names aren't right, I'm not sure what I should be using...

Thank you!
